### PR TITLE
[Enhancement] support session var partial_update_mode

### PR DESCRIPTION
**Pull request by luohaha**
## Problem Summary:
Fixes #20436
Support session var `partial_update_mode`. `partial_update_mode` support three types now:
1. `auto`. It means that `UPDATE` syntax will use the default strategy to decide whether to perform partial column update:
* No WHERE condition is used.
* The percentage of updated columns to all columns is less than 30% and the number of updated columns is less than 4.
2. `column`.  It means that `UPDATE` syntax will use partial update by column.
3. `row`. It means that `UPDATE` syntax will use full columns update.

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 12 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[941, 942]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|6|8|75.00%|[92, 108]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java [hunk 1]
```diff
@@ -87,14 +87,25 @@ public static void analyze(UpdateStmt updateStmt, ConnectContext session) {
...
 
-        if (updateStmt.getWherePredicate() == null) {
-            if (checkIfUsePartialUpdate(assignmentList.size(), table.getBaseSchema().size())) {
-                // use partial update if:
-                // 1. Columns updated are less than 4
-                // 2. The proportion of columns updated is less than 30%
-                // 3. No where predicate in update stmt
-                updateStmt.setUsePartialUpdate();
-            } else {
+        if (session.getSessionVariable().getPartialUpdateMode().equals("column")) {
```
Comment (decster) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java:90]: column, auto and row should be enum or const?

Comment (luohaha) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java:90]: SessionVar not support enum, so add `PartialUpdateModeConverter` to check if it is column, auto or row.

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 18 (61.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|4|50.00%|[941, 942]|
|:large_blue_circle:|com/starrocks/qe/VariableVarConverters.java|3|6|50.00%|[61, 62, 64]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|6|8|75.00%|[92, 108]|

**Review by decster — APPROVED** [review]

**Pull request merged**